src/io/save_gals_hdf5.c

```diff
@@ -0,0 +1,1149 @@
+                                          "1.0e10 Msun/h", "1.0e10 Msun/h", "1.0e10 Msun/h", "Msun/yr", "Msun/yr", "Msun/yr",
+                                          "Msun/yr", "DiskScaleRadius?", "Cooling?", "Heating?", "1.0e10 Msun/h",
+                                          "Myr", "Myr", "Msun/yr", "1.0e10 Msun/yr", "infallVvir?", "infallVmax?"};
+
+    // These are the HDF5 datatypes for each field.
+    hsize_t tmp_dtype[NUM_OUTPUT_FIELDS] = {H5T_NATIVE_INT, H5T_NATIVE_INT, H5T_NATIVE_LLONG, H5T_NATIVE_LLONG, H5T_NATIVE_INT, 
+                                         H5T_NATIVE_INT, H5T_NATIVE_LLONG, H5T_NATIVE_INT, H5T_NATIVE_INT, 
+                                         H5T_NATIVE_INT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_INT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT,
+                                         H5T_NATIVE_FLOAT, H5T_NATIVE_FLOAT};
+
+    for(int32_t i = 0; i < NUM_OUTPUT_FIELDS; i++) {
+        field_names[i] = tmp_names[i];
+        field_descriptions[i] = tmp_descriptions[i]; 
+        field_units[i] = tmp_units[i];
+        field_dtypes[i] = tmp_dtype[i];
+    }
+
+    return EXIT_SUCCESS;
+}
+
+
+#define TREE_MUL_FAC        (1000000000LL)
+#define THISTASK_MUL_FAC      (1000000000000000LL)
+
+// Take all the properties of the galaxy `*g` and add them to the buffered galaxies
+// properties `save_info->buffer_output_gals`.
+int32_t prepare_galaxy_for_hdf5_output(int32_t treenr, struct GALAXY *g, struct save_info *save_info,
+                                       int32_t output_snap_idx,  struct halo_data *halos, struct halo_aux_data *haloaux,
+                                       struct GALAXY *halogal, const struct params *run_params)
+{
+
+    int32_t filenr = run_params->ThisTask;
+
+    int64_t gals_in_buffer = save_info->num_gals_in_buffer[output_snap_idx];
+
+    save_info->buffer_output_gals[output_snap_idx].SnapNum[gals_in_buffer] = g->SnapNum;
+
+    if(g->Type < SHRT_MIN || g->Type > SHRT_MAX) {
```

Comment:
This check is only relevant if `Type` is being split into `Type` and `isFlyby`

Reply:
Should we be splitting `Type` into `Type` and `isFlyby` by default?  Or should we just remove all instances of this splitting?

Reply:
I am unsure. The `isFlyby` flag is only relevant when we had to alter the input trees to get `sage` to process the halos. If the input trees are already in the original `lhalotree` format then there can not be any flybys (by construction). However, if the input trees in the `lhalotree` format are generated by [ConvertCtrees](https://github.com/manodeep/ConvertCTrees), or if the input trees come from `Genesis`/`Consistent-Trees` formats, then we might want to signal that the input trees were artificially altered. Another option, is to simply change the type while writing the output galaxies - but then that would hide how `sage` processed those galaxies. May be @darrencroton wants to weigh-in on this..